### PR TITLE
Fix GraphTrack overdraw

### PR DESCRIPTION
**Pull request by pierricgimmig**
Redraw curves only when an unpdate is required (zoom, pan), not
on mouse move. The labels do need to be redrawn on mouse move so
split rendering into "UpdatePrimitives" and "Draw".

### CODE REVIEW

File: OrbitGl/GraphTrack.cpp [hunk 1]
```diff
@@ -9,96 +9,84 @@
...
       previous_time = time;
       last_normalized_value = normalized_value;
     }
   }
 }
 
-void GraphTrack::DrawSquareDot(GlCanvas* canvas, Vec2 center, float radius, float z, Color color) {
+void GraphTrack::Draw(GlCanvas* canvas, PickingMode picking_mode, float z_offset) {
+  const bool picking = picking_mode != PickingMode::kNone;
+  if (picking) {
```
Comment (karupayun) [inline OrbitGl/GraphTrack.cpp:67]: Please put the condition inside the if (as we now are using it only once)

if (picking_mode != PickingMode::kNone) ...

**Review by karupayun — APPROVED** [review]
LGTM! Thanks for the change Pierric!

File: OrbitGl/GraphTrack.cpp (hunk 1 above)
Comment (karupayun) [inline OrbitGl/GraphTrack.cpp:67]: Maybe I'd prefer to have the condition inside the if statement, what do you think?

Or you like to have the variable in terms of readability?

Comment (pierricgimmig) [reply OrbitGl/GraphTrack.cpp:67]: It is for readability yes, I'll leave it because I do think it's easier to understand.

**Pull request merged**